### DISCUSSION TITLE
cartCreate and international pricing not persisted as expected through to checkout

### DISCUSSION THREAD

**Discussion by njbmartin** [Shopify/storefront-api-feedback#29] [Help]
![image](https://user-images.githubusercontent.com/5699471/130072558-4986e673-fa58-4fd9-ba60-3a98d70ecd28.png)

We are in the process of supporting multi-currency subscriptions, and would like to "initiate" the checkout in a supported currency. Our default currency is GBP, but we would like to avoid confusing our upcoming US customers by showing them USD as soon as they reach checkout.

Using the API, we can see that the `buyerIdentity` is correctly being set, and the `estimatedCost` is being returned in the right currency and the fixed price we set, however, when redirecting to the checkoutURL provided, the checkout shows the default GBP until the customer sets their address as "United States".

This definitely appears to be a bug, as the documentation suggests it should be persisted:
https://shopify.dev/custom-storefronts/products/international-pricing#create-a-cart

**Comment by njbmartin** [comment]
In comparison, using `checkoutCreate` with `buyerIdentity`, the customer will see the checkout load in the currency set.

**Reply by morganmccunn** [reply]
We're working on a fix!

**Reply by njbmartin** [reply]
Great thank you! Do let me know as soon as I can test this again :)

**Reply by haeky** [reply]
Hello 👋,

This should now be fixed and you should be able to proceed to checkout with the same currency as in the cart. 

Thanks for flagging this!

**Reply by njbmartin** [reply]
Amazing! Can confirm 👍 

![image](https://user-images.githubusercontent.com/5699471/131001358-4b04f8be-95be-4af2-96fc-dc6c2ad2f911.png)

**Reply by njbmartin** [reply]
Is there a particular reason why `checkoutUrl` returns a `cart` URL?
